### PR TITLE
Feature/helloworld_ci

### PR DESCRIPTION
**Pull request by premsekharponnada**
## Description
_This project uses CMake to build a simple "Hello world" cpp projects. The output of the project file was tested by writing a unit test case using gtest (test/test.cc file) which tests the output string to Hello World._

_Use the following commands to run the app in your local machine._

`git clone https://github.com/premsekharponnada/cpp_intro_project.git`
`cd cpp_intro_project`
`cmake -S . -B build` to build files 
`cmake --build build` to make bin files
`ctest` to run tests.

To have support for CircleCI, I had to include gtest submodule in the repository so that circleCI can run gtest. 

I had included the static code analyser Codacy for the repo as a status check before merging.

For more learning on how to implement a basic cpp projects -- [click here][cpp-url]

[cpp-url]: https://levelup.gitconnected.com/how-to-combine-c-cmake-googletest-circleci-docker-and-why-e02d76c060a3

### Automated test cases added
- _Added a gtest test case using `EXPECT_EQ` to compare the equality of output of main file to Hello World._

### Manual test cases run
- _None_

### CODE REVIEW

File: test/test.cc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#include <gtest/gtest.h>
+#include "../src/helloworld.cpp"
```
Comment (anujism) [inline test/test.cc:2]: I think it's better to have absolute import

File: src/helloworld.cpp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+using namespace std;
+string welcome()
+{
+    std::cout << "Hello World" << std::endl;
```
Comment (anujism) [inline src/helloworld.cpp]: why cout? I think we may want some logging later for this but for now, we don't want to print the text.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:focal
```
Comment (anujism) [inline Dockerfile]: I think basic idea for CI/CD should be that you should have a build stage which builds (only if the Dockerfile is changed). and then the next stage builds would use the previous build and build for source and use that to run tests.

**Comment by anujism** [comment]
you don't need `.DS_Store` I think.

**Comment by jjalan** [comment]
@anujism - Does the test setup looks ok? It seems its using both gtest and ctest and one of them seems redundant?

**Comment by anujism** [comment]
> @anujism - Does the test setup looks ok? It seems its using both gtest and ctest and one of them seems redundant?

yeah that raised my eyebrows. I initially thought that gtest is for some magic with tbricks but it was not so I realized tbricks environment is not possible here maybe.

**Pull request merged**